File: 2021/05/27/main.py
class Solution:
    def maxProduct(self, words: List[str]) -> int:
        r = 0
        for i in range(len(words) - 1):
            a = words[i]
            la = len(a)
            for j in range(i+1, len(words)):
                b = words[j]
                lb = len(b)
                for w in a:
                    if w in b:
                        lb = 0
                        break
                r = max(r, la * lb)
        return r

File: 2021/05/27/main.go
package main

import "strings"

func maxProduct(words []string) int {
	r := 0
	for i := 0; i < len(words)-1; i++ {
		a := words[i]
		la := len(a)
		for j := i + 1; j < len(words); j++ {
			b := words[j]
			lb := len(b)
			for k := 0; k < la; k++ {
				if strings.ContainsAny(b, string(a[k])) {
					lb = 0
					break
				}
			}
			if la*lb > r {
				r = la * lb
			}
		}
	}
	return r
}
